openc3/lib/openc3/models/plugin_model.rb

```diff
@@ -68,11 +68,12 @@ def self.all(scope: nil)
     # Doesn't actaully create the plugin during the phase
     def self.install_phase1(gem_file_path, existing_variables: nil, existing_plugin_txt_lines: nil, process_existing: false, scope:, validate_only: false)
       gem_name = File.basename(gem_file_path).split("__")[0]
+      @needs_dependencies = false
```

Comment:
You can't make this an instance variable and use it across install phase1 and phase2.   There could be other installs between that, and they could be in different worker processes.    It may just need to have its own config file scan just to look for NEEDS_DEPENDENCIES and ignore everything else.